muspinsim_gen/muspinsim_gen.xml

```diff
@@ -0,0 +1,116 @@
+<tool id="muspinsim_gen" name="MuSpinSim Generator" version="@TOOL_VERSION@+galaxy@WRAPPER_VERSION@" python_template_version="3.5" profile="22.05" license="MIT">
+    <description>Generate MuSpinSim config from a structure file</description>
+    <macros>
+        <!-- version of underlying tool (PEP 440) -->
+        <token name="@TOOL_VERSION@">2.2.0</token>
+        <!-- version of this tool wrapper (integer) -->
+        <token name="@WRAPPER_VERSION@">0</token>
+        <!-- citation should be updated with every underlying tool version -->
+        <!-- typical fields to update are version, month, year, and doi -->
+        <token name="@TOOL_CITATION@">
+            @software{muspinsim,
+                author = {Sturniolo, Simone and Liborio, Leandro and Owen, Josh and Mudaraddi, Anish and Davies, Joel and Wilkinson, John and {Muon Spectroscopy Computational Project}},
+                license = {MIT},
+                title = {{muspinsim}},
+                url = {https://github.com/muon-spectroscopy-computational-project/muspinsim},
+                version = {v2.2.0},
+                month = {3},
+                year = {2023}
+                doi = {10.5281/zenodo.7704635}
+            }
+        </token>
+    </macros>
+    <creator>
+        <person givenName="Joel" familyName="Davies" url="https://github.com/joelvdavies" identifier="https://orcid.org/0000-0002-4153-6819"/>
+        <organization url="https://muon-spectroscopy-computational-project.github.io/index.html" name="The Muon Spectroscopy Computational Project"/>
+    </creator>
+    <requirements>
+        <requirement type="package" version="@TOOL_VERSION@">muspinsim</requirement>
+    </requirements>
+    <command detect_errors="exit_code"><![CDATA[
+        structure_name_internal=\$(sed 's/ /\_/g' <<< '$structure_file.name') &&
+        ln -s '$structure_file' \$structure_name_internal &&
+        muspinsim-gen \$structure_name_internal $number_closest --out muspinsim_gen_out.in
+        #if $dipolar:
+            --dipolar
+        #end if
+        #if $quadrupolar_conditional.quadrupolar:
+            --quadrupolar
+            #if $quadrupolar_conditional.gipaw_out_file:
+                $quadrupolar_conditional.gipaw_out_file
+            #end if
+        #end if
+        #if $include_interatomic:
+            --include_interatomic
+        #end if
+        --muon_symbol $muon_symbol
+        #for $ignored_symbol in $ignored_symbols:
+        --ignore_symbol $ignored_symbol.ignored_symbol
+        #end for
+        --max_layer $max_layer
+    ]]></command>
+    <inputs>
+        <param type="data" name="structure_file" format="cif,cell,magres" label="Structure file (.cif, .cell or .magres)"/>
```

Comment:
Also, we can presumably also read `extxyz` but NOT `xyz` due to the lack of lattice vectors, needed for expansion of the cell? I tried with a random xyz I had and got `ValueError: arange: cannot compute length`, which I then didn't get for an extxyz that was otherwise the same.

Both would be captured by making it accept `txt`, or we could add `extxyz` to the list.

Reply:
Yes that's right, I would rather allow datatypes that we have tested than allow all. I guess we should either allow all of them and create an issue to change this and use it `struct.$structure_file.extension` as below once magres has been added, or just wait. I think this also depends on what is involved in adding the datatype, I assume it wouldn't work until the next galaxy release? I can give it a bit of time and see if I get a chance to have a proper look, at the moment I have other things to do as well.

Reply:
I remember Eli saying that addition of new datatypes can target the latest existing release branch, but looking at https://github.com/galaxyproject/galaxy/blob/dev/CONTRIBUTING.md I can't actually see it mentioned...